.github/workflows/node.js.yml

```diff
@@ -0,0 +1,107 @@
+        uses: actions/checkout@v3
+
+      - name: Node.js setup
+        uses: actions/setup-node@v3
+        with:
+          node-version: '16.x'
+
+      - name: 📦 Install dependencies
+        run: yarn install
+
+      - name: 💅 Prettier check
+        run: yarn prettier:check
+
+      - name: 💨 Lint check
+        run: yarn lint
+
+      - name: 🏗️ Build app
+        run: yarn ng build --prod
+
+  Deploy:
+    if: ${{ github.event_name == 'push' && (github.ref == 'refs/heads/master' || github.ref == 'refs/heads/staging')}}
+    runs-on: ubuntu-latest
+    environment:
+      name: ${{ (github.ref == 'refs/heads/master' && 'prod') || 'stage' }}
+    needs: PreDeployTesting
+    permissions:
+      id-token: write
+      contents: read
+
+    steps:
+      - name: Repository checkout
+        uses: actions/checkout@v3
+
+      - name: Install AWSCLI
+        run: pip install --user awscli
+
+      - name: Configure AWS credentials
+        uses: aws-actions/configure-aws-credentials@036a4a1ddf2c0e7a782dca6e083c6c53e5d90321
+        with:
+          aws-region: us-east-1
+          role-to-assume: ${{ secrets.AWS_IAM_ROLE_ARN }}
+
+      - name: Node.js setup
+        uses: actions/setup-node@v3
+        with:
+          node-version: '16.x'
+
+      - name: 📦 Install dependencies
+        run: yarn install
+
+      - name: 🏗️ Build app
+        run: yarn ng build --prod
+
+      - name: Copy embed.js
+        run: cp embed/embed.js dist/embed.js
+
+      - name: Copy example.html
+        run: cp embed/example.html dist/example.html
+
+      - name: Copy mobile directory
+        run: cp -r mobile/ dist/mobile/
+
+      - name: Update embed host (staging)
+        if: github.ref == 'refs/heads/staging'
+        run: sed -i "s%$appDomainPlaceholder%$appDomain%g" dist/embed.js
+        env:
+          appDomainPlaceholder: '{appDomain}'
+          appDomain: 'https://stage.knowgod.com'
+
+      - name: Deploy to s3 (staging)
+        if: github.ref == 'refs/heads/staging'
+        run: |
+          aws s3 sync dist/knowgod s3://$S3_BUCKET --acl 'public-read' --exclude '*.html' --cache-control 'public, max-age=31536000' --delete
+          aws s3 sync dist/knowgod s3://$S3_BUCKET --acl 'public-read' --include '*.html' --cache-control 'no-cache' --delete
+          aws s3 cp dist/embed.js s3://$S3_BUCKET --acl 'public-read' --cache-control 'no-cache'
+          aws s3 cp dist/example.html s3://$S3_BUCKET --acl 'public-read' --cache-control 'no-cache'
+          aws s3 sync dist/mobile s3://$S3_BUCKET --acl 'public-read' --cache-control 'no-cache' --content-type 'application/json'
+        env:
+          S3_BUCKET: ${{ secrets.AWS_S3_BUCKET }}
+
+      - name: Deploy prod
+        if: github.ref == 'refs/heads/master'
+        run: echo "Deploy code goes here."
+
+      - name: Deploy summary
+        run: echo "Deployed to the ${{ (github.ref == 'refs/heads/master' && 'prod') || 'stage' }} environment from the ${{ github.ref_name }} branch."
```

Comment:
This step is rather pointless, and the message will be buried in the action run log which would pretty quickly show if the deploy succeeded or failed anyways

Reply:
This is essentially a `console.log` for testing. It'll be removed before going live.